### ISSUE TITLE
Server-side cropping for pre-1.8 textures

### ISSUE THREAD

**Issue by dreamscached** [elyby/chrly#42]
Hi, and first thank you for this convenient service. 

I have a proposal: versions before 1.8 use 64x32 skins and expect the API endpoint to return 64x32 texture, failing to properly render it in case a 64x64 skin is returned.

To address that, I suggest cropping images before serving them to the client. This is as simple as just removing bottom 64x32 part of the image, additionally it can be possible to fix slim texture by simply mirroring one pixel wide strip on the arms so it doesn't cause black areas on the skin.

If you approve the idea, I'm ready to work on the necessary changes and submit a PR.

**Comment by erickskrauch** [comment]
Hello. Thank you for the issue. This is a really useful functionality, but we're unable to implement it.

Chrly v5 is currently under development. It will be an API that provides access purely and only to meta information about textures, but doesn't manage them. This means that Chrly knows nothing about the final skin data and cannot change it.

These changes should be done on the Ely.by side, but to do it there, we need to know exactly from which version of the game the textures are requested. Unfortunately, the game client doesn't provide such information.

So, in summary, I can only state that we actually have no possibility to implement this functionality :(

**Comment by dreamscached** [comment]
Speaking of Ely.by, the documentation mentions the use of `minecraft_version` query parameter and it's present in jarmods available for download, however I didn't notice any change of output data no matter what version is provided. Is it outdated?

**Comment by erickskrauch** [comment]
I have to admit, I'd forgotten about that parameter myself. This parameter is indeed present in reference implementations. We originally added it only for analytics purposes, but it can indeed be used for skin conversion purposes as well.

However, this is still an extremely complex task, as we would have to literally put a handler on every skin request to make a decision depending on whether the skin should be degraded or not. We may do this in the future, but at the moment the load on Ely.by doesn't allow us to do it in an "easy way". And the lack of time resources doesn't let us to do it in a "complex way".

**Comment by dreamscached** [comment]
Got it, thank you for your explanations!
